components/DetailsPane.vue

```diff
@@ -0,0 +1,93 @@
+<script lang="ts">
+import { defineComponent } from "vue";
+
+export default defineComponent({
+    props: {
+        initOpen: {
+            type: Boolean,
+            default: false
+        },
+        initDetails: {
+          type: Object,
+          default () {
+            return {};
+          }
+        },
+        initName: {
+          type: String,
+          default: ""
+        }
+    },
+    data () : Object {
+        return {
+            open: this.initOpen,
+            details: this.initDetails,
+            name: this.initName
+        };
+    },
+    mounted () {
+      /**
+       * event triggered on toggling the details window
+       * @param name the name of the tracker to show in the details pane
+       * @param trackerDetails the tracker details Object (key-value pairs)
+       */
+        this.$root?.$on("details", (name : string, trackerDetails : Object) => {
```

Comment:
Instead of typing the trackerDetails with just `Object`, you could make an interface that describes what the object contains, which will let TypeScript better autocomplete.

Reply:
Yes, we can definitely do this. Not sure if it is needed for this PR, since the format of the trackerDetails might still change when we get the actual data. It's your call, if you want me to make a dummy interface already or not

Reply:
True, but write a to do there, so that we don't forget it